gradebookng/impl/src/main/java/org/sakaiproject/grading/impl/GradingServiceImpl.java

```diff
@@ -4775,18 +4784,24 @@ public CommentDefinition getAssignmentScoreComment(String gradebookUid, Long ass
         if (StringUtils.isBlank(gradebookUid) || assignmentId == null || StringUtils.isBlank(studentUid)) {
             throw new IllegalArgumentException("gradebookUid, assignmentId and studentUid must be valid.");
         }
-
+        String assignmentName = "";
         final GradebookAssignment assignment = getAssignmentWithoutStats(gradebookUid, assignmentId);
         if (assignment == null) {
-            throw new AssessmentNotFoundException("There is no assignmentId " + assignmentId + " for gradebookUid " + gradebookUid);
+            CourseGrade courseGrade = getCourseGrade(getGradebook(gradebookUid).getId());
+            if(courseGrade != null && courseGrade.getId().equals(assignmentId)){   //check if this is a course grade before declaring it Not Found
```

Comment:
I'd like to see a unit test to cover the changes in this PR. There's a lot more testing in the grading service after my big refactor, so it's easier to add them

Reply:
I added one [GradingServiceTests.courseGradeComment] closely following the example of your assignmentScoreComment test. Please let me know of any others you'd like added!